jib-core/src/test/java/com/google/cloud/tools/jib/filesystem/FileOperationsTest.java

```diff
@@ -44,9 +45,10 @@ private static void verifyWriteWithLock(Path file) throws IOException {
     OutputStream fileOutputStream = FileOperations.newLockingOutputStream(file);
 
     // Checks that the file was locked.
-    try {
-      FileChannel.open(file, StandardOpenOption.READ).tryLock(0, Long.MAX_VALUE, true);
-      Assert.fail("Lock attempt should have failed");
+    try (FileChannel channel = FileChannel.open(file, StandardOpenOption.READ)) {
+      // locking should either fail and return null or throw an OverlappingFileLockException
+      FileLock lock = channel.tryLock(0, Long.MAX_VALUE, true);
```

Comment:
What if we do `lock()` instead of `tryLock()`?

Reply:
It would result in a deadlock.  This `tryLock()` should fail since the stream returned by `FileOperations.newLockingOutputStream()` should be locked.